### PR TITLE
Consolidate CI code format checks in python script

### PR DESCRIPTION
**Pull request by mark-lunarg**
This PR replaces `check_code_format.sh `and `check_commit_message_format.sh` with a new python script, `check_code_format.py`.  

Significant changes include:

- Commit message format errors will now cause CI failures
- script can be run outside of Travis allowing local format checking
- script can function in Windows and Linux
- script can be run from the directory root or from the scripts directory
- script includes more descriptive error and help information

 
**Sample Output with a bunch of test commits to cause errors:**

```
Checking source code and commit formats between TRAVIS_BRANCH=master and TRAVIS_PULL_REQUEST_BRANCH=markl-commit-msgs-testing
 
Checking PR source code for clang-format errors: 
 
Found formatting errors! 
 
--- layers/buffer_validation.cpp	(before formatting)
+++ layers/buffer_validation.cpp	(after formatting)
@@ -359,27 +359,35 @@
 }
 
 bool IMAGE_VIEW_STATE::OverlapSubresource(const IMAGE_VIEW_STATE &compare_view) const {
-    if (image_view == compare_view.image_view) { return true; }
-    if (image_state->image != compare_view.image_state->image) { return false; } if (normalized_subresource_range.aspectMask != compare_view.normalized_subresource_range.aspectMask) { return false;
+    if (image_view == compare_view.image_view) {
+        return true;
+    }
+    if (image_state->image != compare_view.image_state->image) {
+        return false;
+    }
+    if (normalized_subresource_range.aspectMask != compare_view.normalized_subresource_range.aspectMask) {
+        return false;
     }
 
     // compare if overlap mip level
     if ((normalized_subresource_range.baseMipLevel < compare_view.normalized_subresource_range.baseMipLevel) &&
         ((normalized_subresource_range.baseMipLevel + normalized_subresource_range.levelCount) <=
-         compare_view.normalized_subresource_range.baseMipLevel)) { return false;
+         compare_view.normalized_subresource_range.baseMipLevel)) {
+        return false;
     }
 
     if ((normalized_subresource_range.baseMipLevel > compare_view.normalized_subresource_range.baseMipLevel) &&
         (normalized_subresource_range.baseMipLevel >=
          (compare_view.normalized_subresource_range.baseMipLevel + compare_view.normalized_subresource_range.levelCount))) {
-        return false; }
+        return false;
+    }
 
     // compare if overlap array layer
     if ((normalized_subresource_range.baseArrayLayer < compare_view.normalized_subresource_range.baseArrayLayer) &&
         ((normalized_subresource_range.baseArrayLayer + normalized_subresource_range.layerCount) <=
          compare_view.normalized_subresource_range.baseArrayLayer)) {
         return false;
-    		}
+    }
 
     if ((normalized_subresource_range.baseArrayLayer > compare_view.normalized_subresource_range.baseArrayLayer) &&
         (normalized_subresource_range.baseArrayLayer >=
 
 
Checking PR source files for correct copyright information: 
 
The modified source files have correct copyright dates.
 
 
Checking PR commit messages for consistency issues: 
 The following subject line must start with a single word specifying the functional area of the change, followed by a colon and space. 
 e.g., 'layers: Subject line here' or 'corechecks: Fix off-by-one error in ValidateFences'. 
 Other common module names include layers, build, cmake, tests, docs, scirpts, stateless, gpu, syncval, practices, etc. 
      'tobin:Wish you were here'
 
 The following subject line exceeds 64 characters in length. 
      'bob: So far this is all looking good, but how long must this error'
 
 For the following commit, the last character of the subject line must not be a period or comma. 
      'fred: Well, no periods here please.'
 
The first word of the subject line after the ':' character must be capitalized. 
      'layers: dangit forgot the big D'
 
 The following subject line must start with a single word specifying the functional area of the change, followed by a colon and space. 
 e.g., 'layers: Subject line here' or 'corechecks: Fix off-by-one error in ValidateFences'. 
 Other common module names include layers, build, cmake, tests, docs, scirpts, stateless, gpu, syncval, practices, etc. 
      'layers this one has no colon'
 
 The first word of the subject line must be lower case. 
      'Layers: First error'
 
 Commit Message Format Requirements: 
 ----------------------------------- 
 o  Subject lines must be <= 64 characters in length 
 o  Subject lines must start with a module keyword which is lower-case and followed by a colon and a space 
 o  The first word following the colon must be capitalized and the subject line must not end in a '.' 
 o  The subject line must be followed by a blank line 
 o  The commit description must be <= 72 characters in width
 
 Examples: 
 --------- 
      build: Fix Vulkan header/registry detection for SDK 
      tests: Fix QueryPerformanceIncompletePasses stride usage 
      corechecks: Fix validation of VU 03227 
      state_tracker: Remove 'using std::*' statements 
      stateless: Account for DynStateWithCount for multiViewport
 
 Refer to this document for additional detail: 
 https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/CONTRIBUTING.md#coding-conventions-and-formatting 
 
One or more format checks failed.
```

### CODE REVIEW

File: scripts/check_code_format.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    HELP_MSG ='\033[0;36m'
+    SUCCESS_MSG ='\033[1;32m'
+    CONTENT ='\033[1;39m'
+    ERR_MSG ='\033[1;31m'
+    NO_COLOR ='\033[0m'
+
+#
+# Color print routine, takes a string matching a txtcolor above and the output string, resets color upon exit
+def CPrint(msg_type, msg_string):
+    if msg_type == 'HELP_MSG':
```
Comment (ncesario-lunarg) [inline scripts/check_code_format.py:55]: Should `color` be assigned a default value?

File: scripts/check_code_format.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        color = txtcolors.CONTENT
+    if msg_type == 'ERR_MSG':
+        color = txtcolors.ERR_MSG
+    if msg_type == 'NO_COLOR' or msg_type == '':
+        color = txtcolors.NO_COLOR
+    print(color, msg_string, txtcolors.NO_COLOR)
+
+#
+# Get list of files involved in this branch
+target_files_data = subprocess.check_output(['git', 'diff', '--name-only', 'master'])
```
Comment (ncesario-lunarg) [inline scripts/check_code_format.py:69]: Can `origin/master` be used as the ref for comparison?

**Review by ncesario-lunarg — COMMENTED** [review]
A couple minor nits/questions from trying this out:
1. Does it make sense to use `origin/master` instead of `master` for branch comparison? e.g., when I tried this out I got notified of a commit error about a commit that didn't belong to me because my local checkout of `master` was out of date, but I did have the latest master fetched. I think `origin/master` should allow the user to get the same functionality without having to checkout master and update it.
2. Looks like `check_code_format.py` is meant to be used as an executable (due to `#!/usr/bin/env python`), but the file does not have executable permissions (on linux)?

Here's a [diff](https://github.com/KhronosGroup/Vulkan-ValidationLayers/files/5532207/check_format.diff.gz) of some changes I tried out locally.

File: scripts/check_code_format.py (hunk 1 above)
Comment (mark-lunarg) [reply scripts/check_code_format.py:55]: Yes, will do.

File: scripts/check_code_format.py (hunk 2 above)
Comment (mark-lunarg) [reply scripts/check_code_format.py:69]: Yes, great idea.

**Review by ncesario-lunarg — APPROVED** [review]

File: scripts/check_code_format.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                CPrint('ERR_MSG', "For the following commit, the last character of the subject line must not be a period or comma.")
+                CPrint('CONTENT', "     '" + msg_line_text + "'\n")
+                retval = 1
+            # Output error if subject line doesn't start with 'module: '
+            if 'Revert' not in msg_line_text:
+                module_name = msg_line_text.split(' ')[0]
+                if module_name[-1] != ':':
+                    CPrint('ERR_MSG', "The following subject line must start with a single word specifying the functional area of the change, followed by a colon and space.")
+                    CPrint('ERR_MSG', "e.g., 'layers: Subject line here' or 'corechecks: Fix off-by-one error in ValidateFences'.")
+                    CPrint('ERR_MSG', "Other common module names include layers, build, cmake, tests, docs, scirpts, stateless, gpu, syncval, practices, etc.")
```
Comment (mikes-lunarg) [inline scripts/check_code_format.py:171]: spelling, "scirpts"

File: scripts/check_code_format.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
```
Comment (mikes-lunarg) [inline scripts/check_code_format.py:1]: Worth setting the executable bit on the file?

File: scripts/check_code_format.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+# Check clang-formatting of source code diff
+def VerifyClangFormatSource():
+    CPrint('', "\nChecking PR source code for clang-format errors:")
+    retval = 0
+    if not os.path.isfile('./scripts/clang-format-diff.py'):
+        CPrint('HELP_MSG', "Missing /scripts/clang-format-diff.py, cannot run clang-format checks.")
+        CPrint('HELP_MSG', "The missing script can be installed by running this command in the root of the repository:\n")
+        CPrint('HELP_MSG', "    curl -L http://llvm.org/svn/llvm-project/cfe/trunk/tools/clang-format/clang-format-diff.py -o scripts/clang-format-diff.py")
+    else:
+        good_file_pattern = re.compile('.*\.(cpp|cc|c\+\+|cxx|c|h|hpp)$')
```
Comment (mikes-lunarg) [inline scripts/check_code_format.py:85]: unescaped backslash, either escape it with `\\` or use a raw string

File: scripts/check_code_format.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+    commit_data = check_output(pr_commit_range_parms)
+    commit_text = commit_data.decode('utf-8')
+
+    if commit_text is None:
+        CPrint('SUCCESS_MSG', "\nNo commit messages were found for format checks.\n")
+        return retval
+
+    msg_cur_line = 0
+    msg_prev_line = ''
```
Comment (mikes-lunarg) [inline scripts/check_code_format.py:147]: `msg_prev_line` does not appear to be used for anything

File: scripts/check_code_format.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    SUCCESS_MSG ='\033[1;32m'
+    CONTENT ='\033[1;39m'
+    ERR_MSG ='\033[1;31m'
+    NO_COLOR ='\033[0m'
+
+#
+# Color print routine, takes a string matching a txtcolor above and the output string, resets color upon exit
+def CPrint(msg_type, msg_string):
+    color = txtcolors.NO_COLOR
+    if msg_type == 'HELP_MSG':
```
[the remark this answers is not in the document]
Comment (mark-lunarg) [reply scripts/check_code_format.py:57]: This wasn't in the "Python for old white guys who don't know python" book.

File: scripts/check_code_format.py (hunk 6 above)
Comment (mark-lunarg) [reply scripts/check_code_format.py:147]: Good catch, this got nuked in a copy/paste tragedy.

**Pull request merged**